File: decompiled/sources/com/hjq/permissions/PermissionDelegateImplV26.java
package com.hjq.permissions;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;

class PermissionDelegateImplV26 extends PermissionDelegateImplV23 {
    PermissionDelegateImplV26() {
    }

    public boolean isGrantedPermission(Context context, String str) {
        if (PermissionUtils.equalsPermission(str, Permission.REQUEST_INSTALL_PACKAGES)) {
            return isGrantedInstallPermission(context);
        }
        if (PermissionUtils.equalsPermission(str, Permission.READ_PHONE_NUMBERS) || PermissionUtils.equalsPermission(str, Permission.ANSWER_PHONE_CALLS)) {
            return PermissionUtils.checkSelfPermission(context, str);
        }
        return super.isGrantedPermission(context, str);
    }

    public boolean isPermissionPermanentDenied(Activity activity, String str) {
        if (PermissionUtils.equalsPermission(str, Permission.REQUEST_INSTALL_PACKAGES)) {
            return false;
        }
        if (!PermissionUtils.equalsPermission(str, Permission.READ_PHONE_NUMBERS) && !PermissionUtils.equalsPermission(str, Permission.ANSWER_PHONE_CALLS)) {
            return super.isPermissionPermanentDenied(activity, str);
        }
        if (PermissionUtils.checkSelfPermission(activity, str) || PermissionUtils.shouldShowRequestPermissionRationale(activity, str)) {
            return false;
        }
        return true;
    }

    public Intent getPermissionIntent(Context context, String str) {
        if (PermissionUtils.equalsPermission(str, Permission.REQUEST_INSTALL_PACKAGES)) {
            return getInstallPermissionIntent(context);
        }
        return super.getPermissionIntent(context, str);
    }

    private static boolean isGrantedInstallPermission(Context context) {
        return context.getPackageManager().canRequestPackageInstalls();
    }

    private static Intent getInstallPermissionIntent(Context context) {
        Intent intent = new Intent("android.settings.MANAGE_UNKNOWN_APP_SOURCES");
        intent.setData(PermissionUtils.getPackageNameUri(context));
        return !PermissionUtils.areActivityIntent(context, intent) ? PermissionUtils.getApplicationDetailsIntent(context) : intent;
    }
}

File: decompiled/sources/com/hjq/permissions/PermissionDelegateImplV26.kt
package com.hjq.permissions

import android.app.Activity
import android.content.Context
import android.content.Intent

internal open class PermissionDelegateImplV26 : PermissionDelegateImplV23() {
    override fun isGrantedPermission(context: Context, permission: String): Boolean {
        if (PermissionUtils.equalsPermission(permission, Permission.REQUEST_INSTALL_PACKAGES)) {
            return isGrantedInstallPermission(context)
        }
        if (PermissionUtils.equalsPermission(permission, Permission.READ_PHONE_NUMBERS) ||
            PermissionUtils.equalsPermission(permission, Permission.ANSWER_PHONE_CALLS)
        ) {
            return PermissionUtils.checkSelfPermission(context, permission)
        }
        return super.isGrantedPermission(context, permission)
    }

    override fun isPermissionPermanentDenied(activity: Activity, permission: String): Boolean {
        if (PermissionUtils.equalsPermission(permission, Permission.REQUEST_INSTALL_PACKAGES)) {
            return false
        }
        if (!PermissionUtils.equalsPermission(permission, Permission.READ_PHONE_NUMBERS) &&
            !PermissionUtils.equalsPermission(permission, Permission.ANSWER_PHONE_CALLS)
        ) {
            return super.isPermissionPermanentDenied(activity, permission)
        }
        return !PermissionUtils.checkSelfPermission(activity, permission) &&
            !PermissionUtils.shouldShowRequestPermissionRationale(activity, permission)
    }

    override fun getPermissionIntent(context: Context, permission: String): Intent {
        if (PermissionUtils.equalsPermission(permission, Permission.REQUEST_INSTALL_PACKAGES)) {
            return getInstallPermissionIntent(context)
        }
        return super.getPermissionIntent(context, permission)
    }

    private fun isGrantedInstallPermission(context: Context): Boolean =
        context.packageManager.canRequestPackageInstalls()

    private fun getInstallPermissionIntent(context: Context): Intent {
        val intent = Intent("android.settings.MANAGE_UNKNOWN_APP_SOURCES")
        intent.data = PermissionUtils.getPackageNameUri(context)
        return if (!PermissionUtils.areActivityIntent(context, intent)) {
            PermissionUtils.getApplicationDetailsIntent(context)
        } else {
            intent
        }
    }
}
